bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/HistoricalPersistenceStoreService.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.persistence;
+
+/**
+ * This class provides an interface to the a {@link PersistenceService} to allow data to be stored
+ * at a specific time. This allows bindings that interface to devices that store data internally,
+ * and then periodically provide it to the server to be accommodated.
+ *
+ * @author Chris Jackson - Initial implementation and API
+ *
+ */
+public interface HistoricalPersistenceStoreService {
```

Comment:
> So "Store" is definitely too much here.

Ok - ```HistoricalPersistenceService```?

> How about ModifiablePersistenceService?

But it's not modifyable - it just allows you to store information with a time - you can't update anything in the store so in this respect it's no different than the other services.

> Should this interface directly extend QueryablePersistenceService

Yes - you're probably right - I don't think it makes sense without query.

> The PersistenceAdminService also wants to address removal of entries. If I remember our past discussions right, you probably want to add some delete/remove methods to this interface as well, don't you?

Yes - I split those into a separate interface though. I'm happy to consolidate into a single interface if you prefer, but I recall in the past you wanted to keep admin (delete) functions separate.